model/model_training/custom_datasets/prompt_dialogue.py

```diff
@@ -199,45 +198,37 @@ def __getitem__(self, idx):
 class DolphinMix(Dataset):
     name = "dophin-mix"
 
-    def __init__(self, cache_dir, num_samples=100000, max_char_len=8000, seed=42):
+    def __init__(self, cache_dir, num_samples: Optional[int] = None, max_char_len: int = 8000, seed: int = 42):
```

Comment:
Isn't it better to add `data_files` as an argument with the default value "flan-5m..."?  In this way, we can change to gpt-4 version using config if required. 
`